client/src/js/jobs/components/__tests__/List.test.js

```diff
@@ -0,0 +1,68 @@
+import { JobsList, mapStateToProps, mapDispatchToProps } from "../List";
+
+describe("<JobsList />", () => {
+    let props;
+
+    beforeEach(() => {
+        props = {
+            documents: [{ foo: "bar" }],
+            total_count: 5,
+            page: 2,
+            page_count: 3,
+            term: "foo",
+            onLoadNextPage: jest.fn(),
+            canRemove: jest.fn(),
+            canCancel: jest.fn()
+        };
+    });
+
+    it("should render", () => {
+        const wrapper = shallow(<JobsList {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it("componentDidMount should call onLoadNextPage", () => {
+        expect(props.onLoadNextPage).not.toHaveBeenCalled();
+        shallow(<JobsList {...props} />);
+        expect(props.onLoadNextPage).toHaveBeenCalledWith("foo", 1);
+    });
+
+    it("should render when [this.props.documents === null]", () => {
+        props.documents = null;
+        const wrapper = shallow(<JobsList {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it("should render when [this.props.documents.length === false]", () => {
+        props.documents = {};
+        const wrapper = shallow(<JobsList {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+});
+
+describe("mapStateToProps", () => {
+    const state = {
+        jobs: {
+            term: "foo"
+        },
+        account: {
+            administrator: true,
+            permissions: ["foo"]
+        }
```

Comment:
This does not cover all test cases!

Either provide all test cases or mock out `checkAdminOrPermission()` using `jest.mock()` as we have gone over previously. You will still need to check that `canCancel` and `canRemove` are set to the return value of the mock function.